### ISSUE TITLE
elasticSeach 6 , error : Content-Type header [application/x-www-form-urlencoded] is not supported

### ISSUE THREAD

**Issue by fancyecommerce** [yiisoft/yii2-elasticsearch#166]
Elasticsearch Database Exception – yii\elasticsearch\Exception

```
Elasticsearch request failed with code 406.

Error Info: Array
(
    [requestMethod] => PUT
    [requestUrl] => http://127.0.0.1:9200/fecshop_product_en/_mapping/product
    [requestBody] => {"product":{"properties":{"product_id":{"type":"integer","index":"not_analyzed"},"name":{"type":"text","analyzer":"english"},"short_description":{"type":"text","analyzer":"english"},"description":{"type":"text","analyzer":"english"},"spu":{"type":"keyword","index":"not_analyzed"},"sku":{"type":"keyword","index":"not_analyzed"},"score":{"type":"integer","index":"not_analyzed"},"status":{"type":"integer","index":"not_analyzed"},"is_in_stock":{"type":"string","index":"not_analyzed"},"url_key":{"type":"keyword","index":"not_analyzed"},"price":{"type":"float","index":"not_analyzed"},"cost_price":{"type":"float","index":"not_analyzed"},"special_price":{"type":"intfloateger","index":"not_analyzed"},"special_from":{"type":"integer","index":"not_analyzed"},"special_to":{"type":"integer","index":"not_analyzed"},"final_price":{"type":"float","index":"not_analyzed"},"image":{"type":"text","index":"not_analyzed"},"created_at":{"type":"integer","index":"not_analyzed"},"sync_updated_at":{"type":"float","index":"not_analyzed"}}}}
    [responseCode] => 406
    [responseHeaders] => Array
        (
            [content-type] => application/json; charset=UTF-8
            [content-length] => 97
        )

    [responseBody] => Array
        (
            [error] => Content-Type header [application/x-www-form-urlencoded] is not supported
            [status] => 406
        )

)

```

```

485486487488489490491492493494495496497498499500501502503

                    'requestBody' => $requestBody,
                    'responseCode' => $responseCode,
                    'responseHeaders' => $headers,
                    'responseBody' => $this->decodeErrorBody($body),
                ]);
            }
        } elseif ($responseCode == 404) {
            return false;
        } else {
            throw new Exception("Elasticsearch request failed with code $responseCode.", [
                'requestMethod' => $method,
                'requestUrl' => $url,
                'requestBody' => $requestBody,
                'responseCode' => $responseCode,
                'responseHeaders' => $headers,
                'responseBody' => $this->decodeErrorBody($body),
            ]);
        }
    }


```

**Comment by fancyecommerce** [comment]
my elasticsearch version is 6,

https://www.elastic.co/blog/strict-content-type-checking-for-elasticsearch-rest-requests

**Comment by samdark** [comment]
Are you using 2.1 branch?

**Comment by fancyecommerce** [comment]
my elasticsearch version is 6,
I have solved this problem

Connection.php

377 line

```
CURLOPT_HTTPHEADER     => ['Expect:'],
```

change to:
```
CURLOPT_HTTPHEADER    => [
                'Accept: application/json',
                'Content-Type: application/json'
			],
```

solved!

**Comment by fancyecommerce** [comment]
elasticsearch version 2   does not have this problem

**Comment by fancyecommerce** [comment]
install  yii2 elasticSearch  use 

```

**Comment by mahone2080** [comment]
I also meet this problem today.

**Comment by mahone2080** [comment]
It's better to customize request header in the future.

**Comment by fancyecommerce** [comment]
Waiting for the official to fix this problem   @mahone2080

**Comment by samdark** [comment]
2.0 doesn't support 6.1 as far as I know.

**Comment by fancyecommerce** [comment]
@samdark   2.1 also have problem

**Comment by nikhilvijayanv** [comment]
> CURLOPT_HTTPHEADER    => [
                'Accept: application/json',
                'Content-Type: application/json'
			],

This solves my issue too.

**Comment by siddukrs** [comment]
What is the file path in which the below needs to be placed?

CURLOPT_HTTPHEADER => [
'Accept: application/json',
'Content-Type: application/json'
],

**Comment by kossmoss** [comment]
@siddukrs see vendor/yiisoft/yii2-elasticsearch/Connection.php

**Comment by tarunjangra** [comment]
I can also confirm the same issue. 

```
Array
(
    [requestMethod] => PUT
    [requestUrl] => http://192.168.93.18:9200/items
    [requestBody] => {"settings":{"number_of_shards":3,"number_of_replicas":0},"mappings":{"item":{"properties":{"name":{"type":"text"},"author_name":{"type":"text"},"publisher_name":{"type":"text"},"created_at":{"type":"long"},"updated_at":{"type":"long"},"status":{"type":"long"}}}}}
    [responseCode] => 406
    [responseHeaders] => Array
        (
            [content-type] => application/json; charset=UTF-8
            [content-length] => 97
        )

    [responseBody] => Array
        (
            [error] => Content-Type header [application/x-www-form-urlencoded] is not supported
            [status] => 406
        )

)
```
As suggested by @fancyecommerce, replacing 
```['Expect:']```
with
```[
              'Accept: application/json',
              'Content-Type: application/json'
            ],```
has solved my problem.

@maintainer: When can i expect this to be fixed in main repo?

Thanks

**Comment by cebe** [comment]
This is fixed on the 2.1 branch, see https://github.com/yiisoft/yii2-elasticsearch#requirements
